dev/Projections/Directory.Build.props

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT License. See LICENSE-CODE in the project root for license information. -->
+<!-- This is a special MSBuild file that is parsed before everything else when MSBuild detects it in our directory structure.
+     Adding Sdk="Microsoft.NET.Sdk" to a project causes NuGet files to be generated very early on, so we need these defines to be here
+     to ensure that all of our build output files end up in the same location. -->
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\..\Directory.Build.props" />
+
+  <PropertyGroup>
+    <!-- Both the IID optimizer and the CsWinRT lookup table aren't needed with the new CsWinRT version,
+         so disabling them here until they are disabled automatically by CsWinRT. -->
+    <CsWinRTIIDOptimizerOptOut>true</CsWinRTIIDOptimizerOptOut>
```

Comment:
Is the IIDOptimizer totally obsoleted then, or only when PublishAOT=true?  Is there any downside to leaving it on (e.g., in feeder repos), other than the unused type you mention?

Reply:
yea it is going to be no longer needed in general for newly generated projections at least.  We generate all the IIDs within the projection as we need to expose them as a property in the static methods class.  If someone still had a call to like GuidGenerator in their own code, it can still optimize those if they want to keep it.

Reply:
There is no downside other than not using it anymore is allowing us to fix some of the bugs that came with it such as breaking sourcelink support from rewriting pdbs / dlls.

Reply:
should all feeder projections also be updated?

Reply:
I have similar changes for WinUI.  IXP is not impacted by this as they didn't run IIDOptimizer.  Otherwise, others mainly just Widgets would get these by default in the next CsWinRT update I do, I just didn't want to do that right now.  I decided to do it here manually for now due to the # of projections we produce in this repo.